openstates/tx/votes.py

```diff
@@ -258,55 +242,71 @@ def clean_bill_id(bill_id):
     '[@class = "textpara"]',
     '[contains(translate(., "YEAS", "yeas"), "yeas")]',
 ]
-def record_votes(root, session):
+
+
+def record_votes(root, session, chamber):
     for el in root.xpath('//div{}'.format(''.join(vote_selectors))):
         mv = MaybeVote(el)
         if not mv.is_valid:
             continue
 
-        v = Vote(None, None, 'passage' if mv.passed else 'other', mv.passed,
-                 mv.yeas or 0, mv.nays or 0, mv.present or 0)
-        v['bill_id'] = mv.bill_id
-        v['bill_chamber'] = mv.chamber
-        v['is_amendment'] = mv.is_amendment
-        v['session'] = session[0:2]
-        v['method'] = 'record'
+        v = VoteEvent(
+            chamber=chamber,
+            start_date=None,
+            motion_text='passage' if mv.passed else 'other',
+            result='pass' if mv.passed else 'fail',
+            classification='passage' if mv.passed else 'other',
+            legislative_session=session[0:2],
+            bill=mv.bill_id,
+            bill_chamber=mv.chamber
+        )
+
+        v.set_count('yes', mv.yeas or 0)
+        v.set_count('no', mv.nays or 0)
+        v.set_count('not voting', mv.present or 0)
```

Comment:
`absent` isn't included in the header that gives counts for Yeas and Nays that these other values are pulled from. Could try counting it from the list of names in `mv.votes['absent']`, if you think that's a good idea.